lib/apartment/log_subscriber.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module Apartment
+  class LogSubscriber < ActiveSupport::LogSubscriber
```

Comment:
I think you could inherit from `ActiveRecord::LogSubscriber` instead of the base logger. That way the class would only need to redefine the `sql(event)` method and implement `apartment_log` instead of the whole ActiveRecord logging.